netbox/utilities/views.py

```diff
@@ -142,11 +143,17 @@ def get(self, request):
         if 'pk' in table.base_columns and (permissions['change'] or permissions['delete']):
             table.columns.show('pk')
 
-        # Construct queryset for tags list
-        if hasattr(model, 'tags'):
+        # Add the tags filter field to the from if the model has tags
+        if hasattr(model, 'tags') and filter_form:
             tags = model.tags.annotate(count=Count('extras_taggeditem_items')).order_by('name')
-        else:
-            tags = None
+            choices = [(str(tag.slug), '{} ({})'.format(tag.name, tag.count)) for tag in tags]
+
+            filter_form.fields['tag'] = MultipleChoiceField(
```

Comment:
Love the overall approach (moving tags into a form field), but IMO we should handle this in the form itself rather than in the view. Both because it's a bit cleaner, and to ensure that the tag field is present anywhere else the form might be used.

Not sure if this is better done with a mixin or using a custom form field. We do want to preserve the current behavior of listing only assigned tags; not sure how tricky that will be.

Reply:
The reason I put it in the view is to avoid modifying so much of the code in order to put the `Tags` field at the bottom of the form. If you look at `DeviceFilterForm`, you'll notice that the `field_order` is missing some of the fields, meaning they'll be thrown in the order that they were created at.

If I were class to which the `*FilterForm` will inherit from, I'd have to explicitly define for the rest of the fields in `field_order` such that the `tag` field is placed at the bottom of the search form. If you're okay with me doing that, I'm happy to do it.

Regarding the listing-only-what's-assigned, that's not too much different than what's already happening; don't worry about it.

Reply:
Makes sense.

> I'd have to explicitly define for the rest of the fields in `field_order`

Works for me if you don't mind

Reply:
Alright. I'll get crackin'.

Reply:
Fixed now. Didn't have to play around with the `field_order` after all.